src/main/cpp/CMakeLists.txt

```diff
@@ -270,6 +274,7 @@ target_link_libraries(
     cudf::cudf
     nvtx3::nvtx3-cpp
   -Wl,--no-whole-archive
+    ${ARROW_LIB}
```

Comment:
The fact that we need to add the arrow lib dependency explicitly implies there's a bug in the cudfjni CMakeLists where it should have exported that dependency but did not.

Reply:
The problem is that the cudfjni isn't doing any exporting or installing at all. It is not producing a proper CMake package in the build directory (nor would it if installed, although that's not relevant here since AFAICT it's never installed, just used directly from the build directory). [Note that we're just using `find_library` for the lib instead of `find_package` like we would for something properly set up for export via CMake](https://github.com/NVIDIA/spark-rapids-jni/blob/branch-24.10/src/main/cpp/CMakeLists.txt#L152).